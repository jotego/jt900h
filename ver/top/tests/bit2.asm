    ; jump tests
    main section code
    org 0
    ld a,0xbf    ; common header

    ld xix,data
    bit 0,(xix) ; 0xfe -> bit 0 = 0
    jp ne,bad_end
    or de,1

    bit 1,(xix)
    jp eq,bad_end
    or de,2


end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    end