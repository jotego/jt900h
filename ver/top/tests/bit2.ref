000000bf - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000003 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000035 - 00000000 - 00000000 - 00000000
SR = f804
