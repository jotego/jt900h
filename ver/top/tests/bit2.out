000000bf - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000003 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
00000030 - 00000000 - 00000000 - 00000000
